domain/src/main/java/com/teamagam/gimelgimel/domain/dynamicLayers/remote/SendRemoteAddDynamicEntityRequestInteractor.java

```diff
@@ -17,25 +17,26 @@
   private final DynamicLayersRepository mDynamicLayersRepository;
   private final RetryWithDelay mRetryWithDelay;
   private final GeoEntity mEntity;
+  private final String mDynamicLayerId;
 
   public SendRemoteAddDynamicEntityRequestInteractor(@Provided ThreadExecutor threadExecutor,
       @Provided DynamicLayerRemoteSourceHandler dynamicLayerRemoteSourceHandler,
       @Provided DynamicLayersRepository dynamicLayersRepository,
-      @Provided RetryWithDelay retryWithDelay,
+      @Provided RetryWithDelay retryWithDelay, String dynamicLayerId,
       GeoEntity entity) {
     super(threadExecutor);
     mDynamicLayerRemoteSourceHandler = dynamicLayerRemoteSourceHandler;
     mDynamicLayersRepository = dynamicLayersRepository;
     mRetryWithDelay = retryWithDelay;
+    mDynamicLayerId = dynamicLayerId;
     mEntity = entity;
   }
 
   @Override
   protected SubscriptionRequest buildSubscriptionRequest(DataSubscriptionRequest.SubscriptionRequestFactory factory) {
-    return factory.create(Observable.just(mDynamicLayersRepository),
-        repoObservable -> repoObservable.flatMap(DynamicLayersRepository::getObservable)
+    return factory.create(Observable.just(mDynamicLayerId),
+        dlIdObservable -> dlIdObservable.map(mDynamicLayersRepository::getById)
             .retryWhen(mRetryWithDelay)
```

Comment:
What's the reason for this retrying in the first place?
(If this class had been written in a test-driven manner, I could have gone there looking for documentation :nerd_face:)